### PR TITLE
Recaps and Previews Support

### PR DESCRIPTION
**Pull request by rlauuzo**
## Summary by Sourcery

Add support for identifying and skipping recaps and previews in media files. Introduce new configuration options and UI elements for managing these features. Refactor existing analyzers and database logic to accommodate the new segment types and improve overall performance and maintainability.

New Features:
- Add support for identifying and skipping recaps and previews in media files.
- Introduce new configuration options for chapter detection patterns for recaps and previews.
- Implement new UI elements for configuring auto-skip settings for recaps and previews.

Enhancements:
- Refactor the ChromaprintAnalyzer to improve performance and maintainability by simplifying the logic and reducing unnecessary computations.
- Enhance the database schema and migration logic to support new segment types and improve data integrity.
- Improve the auto-skip functionality to handle multiple segment types and provide better user feedback.

### CODE REVIEW

**Comment by jumoog** [comment]
🥳🥳🥳🥳🥳

**Comment by AbandonedCart** [comment]
This will make things fun. The reason we originally had to check for a second op / intro segment after detecting one by chapter is that a lot of shows slap the same or similar label on both.

**Comment by jumoog** [comment]
looks ready for me

**Comment by jumoog** [comment]
@sourcery-ai review

**Comment by AbandonedCart** [comment]
Why did we remove the option to play part of the credits?

**Review by AbandonedCart — DISMISSED** [review]

**Review by jumoog — DISMISSED** [review]

**Comment by jumoog** [comment]
ready?

**Comment by jumoog** [comment]
@sourcery-ai review

**Comment by rlauuzo** [comment]
> ready?

No!

**Comment by AbandonedCart** [comment]
Wasn’t sure if you meant it to be `action is … || action == …` or if it was a typo, so I went with how it was before to stop the failure.

**Comment by AbandonedCart** [comment]
Auto indent FTW…

**Review by AbandonedCart — APPROVED** [review]

File: IntroSkipper/Data/Segment.cs [hunk 1]
```diff
@@ -2,16 +2,13 @@
...
-using System.Runtime.Serialization;
-using System.Text.Json.Serialization;
 
 namespace IntroSkipper.Data;
 
 /// <summary>
 /// Result of fingerprinting and analyzing two episodes in a season.
 /// All times are measured in seconds relative to the beginning of the media file.
 /// </summary>
-[DataContract(Namespace = "http://schemas.datacontract.org/2004/07/ConfusedPolarBear.Plugin.IntroSkipper.Segment")]
```
Comment (jumoog) [inline IntroSkipper/Data/Segment.cs]: I think we still need this to migrate the XML files.

**Comment by rlauuzo** [comment]
probably needs some testing

**Pull request merged**